### ISSUE TITLE
CTE containing an INSERT returns an error when main SELECT uses a join 

### ISSUE THREAD

**Issue by mkiyama** [greenplum-db/gpdb#7005]
I get an error for this set of tables and CTE. 
I think it should return join of the two tables.

create table bar (a int );
create table foo (a int, c1 text);
insert into bar values (1), (4), (5) ;
insert into foo values (44, 'cas');

WITH my_rows AS (
    insert into foo values (1, 'a'), (2,'b')
    returning *
)
SELECT * FROM my_rows, bar ;
ERROR:  unrecognized node type: 204 (execAmi.c:720)  (seg0 slice3 192.168.180.45:40000 pid=6642) (execAmi.c:720)

centos 6 VM
master and two segment instances on a single host.

running
Greenplum Database 6.0.0 commit:96b332c0dbfc5bbac113565f9e40b8431b574da7-oss)

**Comment by berlin-ab** [comment]
Confirmed. I am able to see the same issue: 

```
± |master → berlin-ab ↑32 ✓| → psql
psql (9.4.20)
Type "help" for help.

adamberlin=# create table bar (a int );
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE
adamberlin=# create table foo (a int, c1 text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE
adamberlin=# insert into bar values (1), (4), (5) ;
INSERT 0 3
adamberlin=# insert into foo values (44, 'cas');
INSERT 0 1
adamberlin=# WITH my_rows AS (
adamberlin(# insert into foo values (1, 'a'), (2,'b')
adamberlin(# returning *
adamberlin(# )
adamberlin-# SELECT * FROM my_rows, bar ;
ERROR:  unrecognized node type: 204 (execAmi.c:720)  (seg0 slice3 192.168.1.194:25432 pid=87401) (execAmi.c:720)
```

**Comment by danielgustafsson** [comment]
This is addressed in #6822

**Comment by danielgustafsson** [comment]
Fixed in master via #6822, closing.
